crates/engine/tree/src/tree/mod.rs

```diff
@@ -325,7 +338,13 @@ where
         outgoing
     }
 
-    async fn run(mut self) {
+    fn run(mut self) {
+        let persistence_state = Arc::new(Mutex::new(PersistenceState {
+            in_progress: false,
+            last_persisted_hash: None,
+            rx: None,
+        }));
```

Comment:
do we even need to mutex/share this?

perhaps we can make this a field of this type and also move the `last_persisted_block_number` in there?

Reply:
> do we even need to mutex/share this?

yep, good point, no need for it

> perhaps we can make this a field of this type and also move the `last_persisted_block_number` in there?

yeah agree, I had already started moving `last_persisted_block_number` there, we have now a `persistence_state` field